include/mbgl/tile/tile_id.hpp

```diff
@@ -30,9 +30,9 @@ class CanonicalTileID {
     CanonicalTileID scaledTo(uint8_t z) const;
     std::array<CanonicalTileID, 4> children() const;
 
-    const uint8_t z;
-    const uint32_t x;
-    const uint32_t y;
+    uint8_t z;
+    uint32_t x;
+    uint32_t y;
```

Comment:
I don't think it's a good idea to expose them publicly. Instead, we should mark them private and use accessor methods that enforce the constraints. Why do these need to be mutable in the first place? It doesn't look like they're changed.

Reply:
We're making them non-const so that that `CanonicalTileID` can be included in sortable data structures. See comment from @jfirebaugh at https://github.com/mapbox/mapbox-gl-native/pull/10436#discussion_r150689342.

Do we want to make them private/non-const and add const getters?

Reply:
`CanonicalTileID` doesn't maintain any class invariants, so I'm not sure they need to be private. If you want a const canonical tile ID, you can use `const CanonicalTileID`.